tests/Frontend/HtmlSpec.hs

```diff
@@ -74,10 +75,11 @@ spec = do
 -- translate form data back to form input
 
 instance PayloadToEnv ProtoIdea where
-    payloadToEnv view (ProtoIdea t (Markdown d) c) = \case
+    payloadToEnv view (ProtoIdea t (Markdown d) c _is) = \case
         ["", "title"]         -> pure [TextInput t]
         ["", "idea-text"]     -> pure [TextInput d]
         ["", "idea-category"] -> pure [TextInput $ selectCategoryValue "idea-category" view c]
+        [""] -> pure [] -- FIXME: Why? We have a pure field which shouldn't be there.
```

Comment:
Sorry, I'm beeing nit-picky now...

- please remove FIXME
- make `payloadToEnv` a function that calls `payloadToEnv'`, and make `payloadToEnv'` the new method of `PayloadToEnv`.
- `payloadToEnv` should handle the `[""]` case, and `payloadToEnv'` is then free to deal only with the interesting cases.
- while you are at it, you can also make `payloadToEnv'` take `"title"` instead of `["", "title"]`.
- write haddock comment for `payloadToEnv` that explains that we suspect that digestive-functors needs to be touched for a nicer solution.

Also, please watch out for alignment.

Thanks for your patience!  (-: